rust/src/router/active_speaker_observer.rs

```diff
@@ -40,20 +40,10 @@ impl Default for ActiveSpeakerObserverOptions {
     }
 }
 
-/// Represents dominant speaker.
-#[derive(Debug, Clone)]
-pub struct ActiveSpeakerObserverDominantSpeaker {
-    /// The audio producer instance.
-    pub producer: Producer,
-}
-
 #[derive(Default)]
 #[allow(clippy::type_complexity)]
 struct Handlers {
-    dominant_speaker: Bag<
-        Arc<dyn Fn(&ActiveSpeakerObserverDominantSpeaker) + Send + Sync>,
-        ActiveSpeakerObserverDominantSpeaker,
-    >,
+    dominant_speaker: Bag<Arc<dyn Fn(&Producer) + Send + Sync>, Producer>,
```

Comment:
This feels like unnecessary breaking change to be honest.

Reply:
The API documentation said this since day 0. There is no reason to emit an `Object` with a `producer` key instead of a simple `Producer`. Nobody has complained and I've told about these changes to the authors of `ActiveSpeakerObserver`.

Reply:
This is a breaking change though. The API looked one way, now it looks the other way. I'll release this as 0.12.0 instead of 0.11.x, but on TS side this will be shipped as supposedly backwards compatible change while it isn't, this is why I think we have an issue here.

Reply:
Also this is Rust code, it was always documented exactly the way code is written because documentation is auto-generated: https://docs.rs/mediasoup/latest/mediasoup/prelude/struct.ActiveSpeakerObserver.html#method.on_dominant_speaker

Reply:
I see your point, but API documentation states that the event is emitted with a single `Producer` argument since the beginning (and that's what I expected when I wrongly reviewed the feature PR). If the code behaves different than the API docs IMHO it's a "bug".

Reply:
It is a bug, but in the docs, not in the code. In Rust case there was no bug to begin with.

Reply:
ok, I'll revert it next week.

Reply:
I have a better option: on Rust side there is no issue in releasing breaking version of the crate. On TypeScript you can set `Producer.producer` field before firing an event. That way code will match docs, but old assumptions will still be maintained for those who relied on it. We can have a TODO to remove this workaround in v4. WDYT?

Reply:
That's too complex and I'd just hate if we add a `producer` field into a `Producer`. These changes have been released less than 6 hours ago, IMHO it's much better to revert them and change/fix the API docs. I'll do tomorrow probably.

Reply:
NPM might have a feature to remove a version of the package

Reply:
Things reverted in https://github.com/versatica/mediasoup/pull/947